thriftbp/client_pool.go

```diff
@@ -82,9 +82,23 @@ type ClientPoolConfig struct {
 	// In most cases, you would want ConnectTimeout to be short, because if you
 	// have problem connecting to the upstream you want to fail fast.
 	//
-	// But SocketTimeout usually needs to be at least your upstream service's p99
-	// latency SLA. If it's shorter than that you are gonna close connections and
-	// fail requests prematurely.
+	// For SocketTimeout, the value you should set depends on whether you set a
+	// deadline to the context object to the client call functions or not.
+	// If ALL your client calls will have a context object with a deadline
+	// attached, then it's recommended to set SocketTimeout to a short value,
+	// as this is the max overhead the client call will take over the set
+	// deadline, in case the server is not-responding.
+	// But if you don't always have a deadline attached to your client calls,
+	// then SocketTimeout needs to be at least your upstream service's p99 latency
+	// SLA. If it's shorter than that you are gonna close connections and fail
+	// requests prematurely.
+	//
+	// It's recommended to make sure all your client call context objects have a
+	// deadline set, and set SocketTimeout to a short value. For example:
+	//
+	//     clientCtx, cancel := context.WithTimeout(ctx, myCallTimeout)
+	//     defer cancel()
+	//     resp, err := client.MyCall(clientCtx, args)
```

Comment:
This is the only part really changed in baseplate.go. Everything else in this pr are just passing ctx around in accompany to the breaking change. The real change is in thrift.